src/components/organisms/NavBarSchedule/NavBarSchedule.tsx

```diff
@@ -135,54 +128,45 @@ export const NavBarSchedule: React.FC<NavBarScheduleProps> = ({
     });
   }, [selectedDayIndex, firstDayOfSchedule, isScheduleTimeshifted]);
 
-  const getEventLocation = useCallback(
-    (locString: string): VenueLocation => {
-      const [venueId, roomTitle] = extractLocation(locString);
-      const venueName = relatedVenues.find((venue) => venue.id === venueId)
-        ?.name;
-      return { venueId, venueName, roomTitle: roomTitle || undefined };
-    },
-    [relatedVenues]
-  );
-
-  const schedule: ScheduleDay = useMemo(() => {
+  const scheduleNG: ScheduleNGDay = useMemo(() => {
     const startOfSelectedDay = addDays(firstDayOfSchedule, selectedDayIndex);
+    const prioritiseEvents = (
+      a: WithVenueId<VenueEvent>,
+      b: WithVenueId<VenueEvent>
+    ) => {
+      if (a.start_utc_seconds !== b.start_utc_seconds) {
+        return a.start_utc_seconds - b.start_utc_seconds;
+      }
+
+      return a.duration_minutes - b.duration_minutes;
+    };
+
     const daysEvents = relatedVenueEvents
       .filter(
         isScheduleTimeshifted
           ? isEventWithinDate(startOfSelectedDay)
           : isEventWithinDateAndNotFinished(startOfSelectedDay)
       )
+      .sort(prioritiseEvents)
       .map(
         prepareForSchedule({
           day: startOfSelectedDay,
           usersEvents: userEventIds,
         })
       );
 
-    const locatedEvents: LocationEvents[] = Object.entries(
-      groupBy(daysEvents, buildLocationString)
-    ).map(([group, events]) => ({
-      events,
-      location: getEventLocation(group),
-    }));
-
     return {
-      locatedEvents,
-      isToday: selectedDayIndex === 0,
-      scheduleDate: startOfSelectedDay,
-      personalEvents: daysEvents.filter((event) => event.isSaved),
+      daysEvents,
     };
   }, [
     relatedVenueEvents,
     userEventIds,
     selectedDayIndex,
-    getEventLocation,
     firstDayOfSchedule,
     isScheduleTimeshifted,
   ]);
 
-  const hasSavedEvents = schedule.personalEvents.length > 0;
+  const hasSavedEvents = scheduleNG.daysEvents.length > 0;
```

Comment:
looks like the meaning of the variable has changed with your update. It was used for the downloading a personalized schedule (if we have saved events -> show button "save my schedule"). But as looks like it's not needed to download the schedule in our schedule v3, we can comment this line. It's not needed anymore.

Reply:
removed download schedule logic. fixed in 03694e555e05deb1888ddfa97d6c16717799922b